### ISSUE TITLE
Streaks when you shoot

### ISSUE THREAD

**Issue by StevenCopeland** [StevenCopeland/SpaceInvaders#2]
When you shoot, the bullet leaves streaks on the screen:

![image](https://user-images.githubusercontent.com/26176516/31036686-64e2ab3e-a53b-11e7-92c8-c43ba20ce345.png)

And that is sort of annoying.

**Comment by nodox** [comment]
You probably need to redraw the canvas. Check the redraw logic.

**Comment by StevenCopeland** [comment]
Really? Here is the function I use for bullets:

```
var shoot = function() {
	ctx.fillStyle = "yellow";
	if (sy >= 0) {
		setInterval(function() {
			ctx.clearRect(x, sy, 10, 10)
			sy -= 7;
			ctx.fillRect(x, sy, 10, 10);
			// if (sy >= mx && sy >= my) {
			// 	ctx.clearRect(mx, my, 30, 30);
			// }
		}, 10)
	} else {
		ctx.clearRect(x, sy, 10, 10)
		sx = x + 0;
		sy = y + -10;
	}
}
```

It runs every time someone shoots(presses the spacebar).

**Comment by creusify** [comment]
Yup, definitely need to doublecheck the redraw logic.

**Comment by StevenCopeland** [comment]
I am not sure what you mean. Do you know what I am doing wrong?

**Comment by nodox** [comment]
@StevenCopeland draw another green rectangle over the area where the yellow bullets are and you will have a quickfix hack.

**Comment by CDeighton** [comment]
Hey there! had a look at this and there was just a small bug where you were clearing the shots using `var x` rather than `var sx`. see #3

**Comment by VRamazing** [comment]
Is this still open or fixed??

**Comment by CDeighton** [comment]
I've made a pull request with a fix (#3), but it was pretty recent so nothing has been merged

**Comment by VRamazing** [comment]
I think the issue arises because you are not updating the screen periodically.Plus there is no structure followed in this code.The code can be improved a lot.If we avoid this step there may be a lot of bugs in the future that may have been avoided

**Comment by CDeighton** [comment]
100% could do it that way. But I think if you go down that path you will find yourself rewriting the majority of the code

**Comment by VRamazing** [comment]
I think it must be done as it will be really difficult to add new features later on.

**Comment by StevenCopeland** [comment]
Okay sorry. I have not been on and have not seen new things. 

This issue is still open.

**Comment by VRamazing** [comment]
Should I work on this and rewrite the code??

**Comment by StevenCopeland** [comment]
I merged a pull request(see #3 ) and that fixed this issue.

**Comment by StevenCopeland** [comment]
But if you want to, you could figure out for me how to just make the box disappear instead of just erasing parts of it.
